File: src/main/java/org/jukeboxmc/block/BlockNetherWart.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemNetherWart;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockNetherWart extends Block {

    public BlockNetherWart() {
        super( "minecraft:nether_wart" );
    }

    @Override
    public ItemNetherWart toItem() {
        return new ItemNetherWart();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.NETHER_WART;
    }

    public void setAge( int value ) {
        this.setState( "age", value );
    }

    public int getAge() {
        return this.stateExists( "age" ) ? this.getIntState( "age" ) : 0;
    }

}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockNetherWart.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.NetherWart
import org.jukeboxmc.api.item.Item
import org.jukeboxmc.api.item.ItemType
import org.jukeboxmc.api.item.enchantment.EnchantmentType
import org.jukeboxmc.server.block.JukeboxBlock
import kotlin.random.Random

class BlockNetherWart(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), NetherWart {

    override fun canPassThrough(): Boolean {
        return true
    }

   override fun getAge(): Int {
       return this.getIntState("age")
   }

   override fun setAge(value: Int): NetherWart {
       return this.setState("age", value)
   }

    override fun getDrops(item: Item): MutableList<Item> {
        if (this.getAge() < 3) {
            return mutableListOf()
        }
        var level = 0
        if (item.hasEnchantment(EnchantmentType.FORTUNE)) {
            level = item.getEnchantment(EnchantmentType.FORTUNE)!!.getLevel()
        }
        return mutableListOf(Item.create(ItemType.NETHER_WART, 2 + Random.nextInt(3) + level))
    }
}